org.eclipse.dawnsci.hdf5.model/src/org/eclipse/dawnsci/hdf5/model/internal/HierarchicalDataFileModel.java

```diff
@@ -87,34 +89,54 @@ private Object getAttribute(String path, String attribute) {
 					return null;
 				}
 				if (node.attributes != null) {
-					return node.attributes.get(attribute);
+					Attribute attr = (Attribute) node.attributes.get(attribute);
+					// attribute does not exist
+					if (attr == null) {
+						return null;
+					}
+					Object attValue = attr.getValue();
+					return NexusUtils.extractScalarFromDataset(attValue);
 				}
 			}
 
 			// attributes have not been loaded
-			try (IHierarchicalDataFile reader = getReader.getReader()) {
-				Map<String, Object> map = reader.getAttributeValues(path);
-				if (map == null) {
-					// path does not exist in file
+			try {
+				NexusFile reader = null;
+				//test if file exists
+				try {
+					reader = getReader.getReader();
+				} catch(NexusException ne) {
+					if (ne.getMessage().startsWith("File not found"))
+						// exit
+						return null;
+				}
+				if (!reader.isPathValid(path)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "reader" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)